runtime/src/qio/sys.c

```diff
@@ -1862,45 +1867,58 @@ err_t sys_socketpair(int domain, int type, int protocol, fd_t* sockfd_out_a, fd_
 extern void chpl_task_yield(void);
 
 err_t sys_select(int nfds, fd_set* readfds, fd_set* writefds, fd_set* exceptfds, struct timeval* timeout, int* nset) {
-
   int got_nset;
   err_t err_out = 0;
-  struct timeval deadline;
-  struct timeval now;
-  struct timeval real_timeout;
-  long wait_usec = 5;
-
-  if (timeout != NULL) {
-    gettimeofday(&deadline, NULL);
-    deadline.tv_sec += timeout->tv_sec;
-    deadline.tv_usec += timeout->tv_usec;
-    if (deadline.tv_usec > 1000000) {
-      deadline.tv_sec++;
-      deadline.tv_usec -= 1000000;
-    }
-  } else if (timeout->tv_sec == 0) {
-    if (timeout->tv_usec < wait_usec)
-      wait_usec = timeout->tv_usec;
+
+  struct timeval first_timeout;
+  struct timeval second_timeout;
```

Comment:
`struct timeval second_timeout = {0}` to zero initialize it. Otherwise it is garbage memory.

Reply:
made the change